test/JDBC/input/BABEL-2795-vu-prepare.sql

```diff
@@ -0,0 +1,64 @@
+CREATE VIEW BABEL_2795_vu_prepare_v1 AS (select datediff(dd,DATETIMEFROMPARTS(2021,12,10,1,50,59,0),DATEADD(minute,123456*1440,DATETIMEFROMPARTS(1900,1,1,0,0,0,0))));
+GO
+
+CREATE VIEW BABEL_2795_vu_prepare_v2 AS (SELECT DATEDIFF(day, '2011-12-29 23:00:00', '2011-12-31 01:00:00'));
+GO
+
+CREATE VIEW BABEL_2795_vu_prepare_v3 AS (SELECT DATEDIFF(dd, '2011-12-30 08:55:59.123', '2011-12-31 08:55:59.123'));
+GO
+
+CREATE VIEW BABEL_2795_vu_prepare_v4 AS (SELECT DATEDIFF(dd, '2011-12-30 08:55:59.123', '2011-12-31 00:00:00.000'));
+GO
+
+CREATE VIEW BABEL_2795_vu_prepare_v5 AS (SELECT DATEDIFF(hour, '2011-12-30 08:55', '2011-12-30 09:05'));
+GO
+
+CREATE PROCEDURE BABEL_2795_vu_prepare_p1 AS (select datediff(dd,DATETIMEFROMPARTS(2021,12,10,1,50,59,0),DATEADD(minute,123456*1440,DATETIMEFROMPARTS(1900,1,1,0,0,0,0))));
+GO
+
+CREATE PROCEDURE BABEL_2795_vu_prepare_p2 AS (SELECT DATEDIFF(day, '2011-12-29 23:00:00', '2011-12-31 01:00:00'));
+GO
+
+CREATE PROCEDURE BABEL_2795_vu_prepare_p3 AS (SELECT DATEDIFF(dd, '2011-12-30 08:55:59.123', '2011-12-31 08:55:59.123'));
+GO
+
+CREATE PROCEDURE BABEL_2795_vu_prepare_p4 AS (SELECT DATEDIFF(dd, '2011-12-30 08:55:59.123', '2011-12-31 00:00:00.000'));
+GO
+
+CREATE PROCEDURE BABEL_2795_vu_prepare_p5 AS (SELECT DATEDIFF(hour, '2011-12-30 08:55', '2011-12-30 09:05'));
+GO
+
+CREATE FUNCTION BABEL_2795_vu_prepare_f1()
```

Comment:
should also call it from procedure and views

Reply:
calling from function, procedure and view